src/app/containers/App/App.jsx

```diff
@@ -1,32 +1,18 @@
-import { useEffect, useState, useRef } from 'react';
+import { useEffect, useLayoutEffect, useState, useRef } from 'react';
 import { renderRoutes } from 'react-router-config';
 import { withRouter } from 'react-router';
-import path from 'ramda/src/path';
 import getRouteProps from '#app/routes/utils/fetchPageData/utils/getRouteProps';
 import usePrevious from '#lib/utilities/usePrevious';
 import getToggles from '#app/lib/utilities/getToggles';
+import routes from '#app/routes';
 
-export const App = ({ routes, location, initialData, bbcOrigin, history }) => {
-  const {
-    service,
-    isAmp,
-    variant,
-    id,
-    assetUri,
-    errorCode,
-    route: { pageType },
-  } = getRouteProps(routes, location.pathname);
+const mapToState = ({ pathname, initialData, routeProps, toggles }) => {
+  const { pageData, status, error, timeOnServer } = initialData;
+  const { service, isAmp, variant, id, assetUri, route } = routeProps;
+  const { pageType } = route;
 
-  const {
-    pageData,
-    toggles,
-    status,
-    error,
-    timeOnServer,
-    showAdsBasedOnLocation,
-  } = initialData;
-
-  const [state, setState] = useState({
+  return {
```

Comment:
Ramda `pick` might be helpful here

It will also ignore anything that doesn't exist, which would prevent the objects having some properties set to undefined like in https://github.com/bbc/simorgh/pull/7825/files#diff-f7adfd5a8540756c182e62628d5613c6R199-R201

Reply:
Yeh I did think about using `pick` 👍

Reply:
@ryanmccombe this look ok? https://github.com/bbc/simorgh/pull/7825/commits/fc12f797f1fcd345af0c6e80855a6789e73170ea

Reply:
👌